lark/tree_matcher.py

```diff
@@ -88,7 +88,7 @@ class TreeMatcher:
 
     def __init__(self, parser):
         # XXX TODO calling compile twice returns different results!
-        assert parser.options.maybe_placeholders == False
+        assert parser.options.maybe_placeholders is False
```

Comment:
Why change this?

If the user wants to give an object that behaves like False, why should we care?

Reply:
I was assuming that this was really looking for the `False` value here, and not something Falsey. If the later is the case it would be better to replace this with `assert not parser.options.maybe_placeholders`, otherwise the user has to write the `__eq__` method in a very specific way.

For context, the [recommendation in pep8](https://peps.python.org/pep-0008/#programming-recommendations) is:

> Don’t compare boolean values to True or False using ==

The `var == False` pattern has a very specific meaning, which is generally not what a user expects. But if that's desired here we can revert it, but based on your comment I think replacing it with  `not var` is the better way to go.